test/e2e/declare_consumer.test.ts

```diff
@@ -98,36 +102,48 @@ describe("declare consumer", () => {
 
   it("declaring a consumer on an existing stream - the consumer should read application properties", async () => {
     const messageApplicationProperties: MessageApplicationProperties[] = []
-    const messages: string[] = []
     const applicationProperties = createApplicationProperties()
     await publisher.send(Buffer.from("hello"), { applicationProperties })
 
     await connection.declareConsumer({ stream: streamName, offset: Offset.first() }, (message: Message) => {
-      messageApplicationProperties.push(message.applicationProperties || {})
-      messages.push("JSON.stringify(message.properties?.correlationId) ||")
+      if (message.applicationProperties) messageApplicationProperties.push(message.applicationProperties)
     })
 
     await eventually(async () => {
       expect(messageApplicationProperties).eql([applicationProperties])
     })
   }).timeout(10000)
+
+  it("declaring a consumer on an existing stream - the consumer should read message annotations", async () => {
+    const messageAnnotations: MessageAnnotations[] = []
+    const annotations = createAnnotations()
+    await publisher.send(Buffer.from("hello"), { messageAnnotations: annotations })
+
+    await connection.declareConsumer({ stream: streamName, offset: Offset.first() }, (message: Message) => {
+      if (message.messageAnnotations) messageAnnotations.push(message.messageAnnotations)
```

Comment:
I don't think this iff is necessary, if you get an error from typescript you can solve it like this
```suggestion
messageAnnotations.push(message.messageAnnotations || {})
```